hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -548,6 +787,31 @@ private Map<String, List<HoodieRecord<HoodieMetadataPayload>>> fetchBaseFileAllR
         .collect(Collectors.groupingBy(Pair::getKey, Collectors.mapping(Pair::getValue, Collectors.toList())));
   }
 
+  private Map<String, HoodieRecord<HoodieMetadataPayload>> fetchBaseFileAllRecordsByPayload(HoodieSeekingFileReader reader,
```

Comment:
Ignore for now. This is again related to the reverse-scanning to aid the writer (to emit tombstone records) - will be change based on the design choice.